akka-http-core/src/main/scala/akka/http/scaladsl/ConnectionContext.scala

```diff
@@ -4,55 +4,151 @@
 
 package akka.http.scaladsl
 
-import akka.stream.TLSClientAuth
-import akka.stream.TLSProtocol._
-import com.typesafe.sslconfig.akka.AkkaSSLConfig
-
-import scala.collection.JavaConverters._
-import java.util.{ Optional, Collection => JCollection }
+import java.util.{ Collections, Optional, Collection => JCollection }
 
 import javax.net.ssl._
 
+import scala.collection.JavaConverters._
 import scala.collection.immutable
 import scala.compat.java8.OptionConverters._
+import scala.util.{ Failure, Success, Try }
+import com.typesafe.sslconfig.akka.AkkaSSLConfig
+import com.typesafe.sslconfig.akka.util.AkkaLoggerFactory
+import com.typesafe.sslconfig.ssl.DefaultHostnameVerifier
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.{ ApiMayChange, InternalApi }
+import akka.stream.{ ConnectionException, TLSClientAuth }
+import akka.stream.TLSProtocol._
 
 trait ConnectionContext extends akka.http.javadsl.ConnectionContext {
   @deprecated("Internal method, left for binary compatibility", since = "10.2.0")
   protected[http] def defaultPort: Int
 }
 
 object ConnectionContext {
-  // ConnectionContext
+  /**
+   *  If you want complete control over how to create the SSLEngine you can use this method:
+   *
+   *  Note that this means it is up to you to make sure features like SNI and Hostname Verification
+   *  are enabled as needed.
+   */
+  @ApiMayChange
+  def https(createSSLEngine: Option[(String, Int)] => () => SSLEngine): HttpsConnectionContext = // ...
```

Comment:
Needs docs about the function argument. Or should we instead provide overloads for` httpsServer` and `httpsClient` so we can get rid of the `Option`?